tests/test_inference/test_takagi_sugeno_inference_system.py

```diff
@@ -0,0 +1,205 @@
+import pytest
+import pandas as pd
+
+from doggos.algebras import GodelAlgebra
+from doggos.inference import TakagiSugenoInferenceSystem
+from doggos.inference.defuzzification_algorithms import weighted_average, takagi_sugeno_karnik_mendel
+from doggos.knowledge import Clause, LinguisticVariable, Domain, fuzzify, Term, Rule
+from doggos.fuzzy_sets import Type1FuzzySet, IntervalType2FuzzySet
+from doggos.knowledge.consequents import TakagiSugenoConsequent
+from doggos.utils.membership_functions import triangular
+
+
+class TestTakagiSugenoInferenceSystem:
+    def test_inference_type_1(self):
+        df = pd.DataFrame({'fire': [100.0, 112, 102.5, 100.1, 103.27],
+                           'air': [1, 4, 3.021, 2.879, 3.55],
+                           'earth': [0, 600, 58, 272, 384],
+                           'water': [0, 13, 5.1, 7.9, 2.2]})
+
+        lingustic_variable_1 = LinguisticVariable('fire', Domain(100, 112.01, 0.01))
+        lingustic_variable_2 = LinguisticVariable('air', Domain(1, 4.001, 0.001))
+        lingustic_variable_3 = LinguisticVariable('earth', Domain(0, 602, 2))
+        lingustic_variable_4 = LinguisticVariable('water', Domain(0, 13.1, 0.1))
+
+        measures = {lingustic_variable_1: [100.0, 112, 102.5, 100.1, 103.27],
+                    lingustic_variable_2: [1, 4, 3.021, 2.879, 3.55],
+                    lingustic_variable_3: [0, 600, 58, 272, 384],
+                    lingustic_variable_4: [0, 13, 5.1, 7.9, 2.2]}
+
+        clause1 = Clause(lingustic_variable_1, 'high', Type1FuzzySet(lambda x: triangular(106., 112., 112.1)(x)))
+        clause2 = Clause(lingustic_variable_1, 'medium', Type1FuzzySet(lambda x: triangular(100, 106., 112)(x)))
+        clause3 = Clause(lingustic_variable_1, 'low', Type1FuzzySet(lambda x: triangular(99.9, 100., 106.)(x)))
+        clause4 = Clause(lingustic_variable_2, 'high', Type1FuzzySet(lambda x: triangular(2.5, 4, 4.1)(x)))
+        clause5 = Clause(lingustic_variable_2, 'medium', Type1FuzzySet(lambda x: triangular(1, 2.5, 4)(x)))
+        clause6 = Clause(lingustic_variable_2, 'low', Type1FuzzySet(lambda x: triangular(0.9, 1, 2.5)(x)))
+        clause7 = Clause(lingustic_variable_3, 'high', Type1FuzzySet(lambda x: triangular(300, 600, 602)(x)))
+        clause8 = Clause(lingustic_variable_3, 'medium', Type1FuzzySet(lambda x: triangular(0, 300, 600)(x)))
+        clause9 = Clause(lingustic_variable_3, 'low', Type1FuzzySet(lambda x: triangular(-2, 0, 300)(x)))
+        clause10 = Clause(lingustic_variable_4, 'high', Type1FuzzySet(lambda x: triangular(6.5, 13, 13.1)(x)))
+        clause11 = Clause(lingustic_variable_4, 'medium', Type1FuzzySet(lambda x: triangular(0, 6.5, 13)(x)))
+        clause12 = Clause(lingustic_variable_4, 'low', Type1FuzzySet(lambda x: triangular(-0.1, 0, 6.5)(x)))
```

Comment:
Fix FuzzySets -> there's no need to use lambda
e.g. Type1FuzzySet(triangular(106., 112., 112.1))